File: Object-Oriented Programming/Produse/Teste.cpp
#include "Teste.h"
#include "Domain.h"
#include "Repository.h"
#include "Service.h"
#include <cassert>

void teste_domain() {

	Produs p{ 1,"a","a",1 };

	assert(p.get_id() == 1);
	assert(p.get_nume() == "a");
	assert(p.get_tip() == "a");
	assert(p.get_pret() == 1);

	Validator val;
	Produs p1{ 1,"","a",111 };
	try {
		val.valideaza(p1);
		assert(false);
	}
	catch (const string&) {
		assert(true);
	}
}

void teste_repo() {

	Repository repo{ "teste.txt" };
	Produs p{ 1,"a","a",1 };
	try {
		repo.adauga_produs(p);
		assert(false);
	}
	catch (const string&) {
		assert(true);
	}
	Produs p1{ 3,"c","c",3 };
	assert(repo.get_all().size() == 2);
	repo.adauga_produs(p1);
	assert(repo.get_all().size() == 3);

	Repository repoErr{ "" };
}

void teste_service() {

	Repository repo{ "teste.txt" };
	Validator val;
	Service srv(repo, val);

	assert(srv.get_all().size() == 2);
	vector<Produs> l = srv.sortare_pret();
	assert(l.at(0).get_pret() == 1);
	assert(l.at(1).get_pret() == 2);

	assert(srv.get_all().size() == 2);
	srv.adauga_produs(3, "d", "d", 3);
	assert(srv.get_all().size() == 3);
}

void ruleaza_teste() {

	teste_domain();
	teste_repo();
	teste_service();
}

File: FundamentalsOfProgramming/Fundamentele Programarii Lucru Suplimentar/FP biciclete/teste.py
from domain.domain import Bicicleta

from repository.repository import Repository,RepositoryException

from service.service import Service

# teste pentru domain
def teste_domain():

    b = Bicicleta(1,"bmx",500.5)
    assert b.getId() == 1
    assert b.getPret() == 500.5
    assert b.getTip() == "bmx"

# teste pentru repository
def teste_repository():

    try:
        f = open("produse_test.txt","w")
    except IOError:
        pass
    
    f.write("100, road, 400.5\n200, road, 400\n300, mountain, 500\n400, bmx, 500\n")
    f.close()
    
    repo = Repository("produse_test.txt")

    l = repo.getAll()
    assert len(l) == 4

    assert repo.cauta_id(100) == 0
    assert repo.cauta_id(1000) == None

    repo.delete(100)
    l = repo.getAll()
    assert len(l) == 3

    assert repo.pret_maxim() == 500
    repo.sterge_biciclete_pret_maxim()
    l = repo.getAll()
    assert len(l) == 1

    try:
        repo.sterge_tip("bicicleta")
        assert False
    except RepositoryException:
        assert True
    
    repo.sterge_tip("road")
    l = repo.getAll()
    assert len(l) == 0
 
# teste pentru service
def teste_service():

    try:
        f = open("produse_test.txt","w")
    except IOError:
        pass
    
    f.write("100, road, 400.5\n200, road, 400\n300, mountain, 500\n400, bmx, 500\n")
    f.close()
    
    repo = Repository("produse_test.txt")
    
    srv = Service(repo)

    l = srv.afisare_bicicelte()
    assert len(l) == 4

    srv.stergere_max()
    l = srv.afisare_bicicelte()
    assert len(l) == 2

    try:
        srv.stergere_tip("dada")
        assert False
    except RepositoryException:
        assert True
    
    srv.stergere_tip("road")
    l = srv.afisare_bicicelte()
    assert len(l) == 0

def ruleaza_teste():

    teste_domain()
    teste_repository()
    teste_service()

ruleaza_teste()